flink-kubernetes-operator/src/main/java/org/apache/flink/kubernetes/operator/crd/spec/FlinkDeploymentSpec.java

```diff
@@ -45,7 +45,7 @@ public class FlinkDeploymentSpec extends AbstractFlinkSpec {
     private String imagePullPolicy;
 
     /** Kubernetes service used by the Flink deployment. */
-    private String serviceAccount;
+    private String serviceAccount = "flink";
```

Comment:
I suggest to leave it on `null` and throw a validation error if not set.

Reply:
thanks for the suggestion. It makes sense to me to add validation check so that serviceAccount will be a required field.
For users who use helm and set a different serviceAccount, they are supposed to be aware of using correct service account.
For users who use helm and use the default serviceAccount, we may add some comments in examples or detailed validation error info to alert users to set the correct serviceAccount(i.e. `flink`).